<a href='https://www.darshan.ac.in/'> <img src='https://www.darshan.ac.in/Content/media/DU_Logo.svg' width="250" height="300"/></a>
<pre>
<center><b><h1>2305CS303 - Python for Data Science</b></center>
<center><b><h1>Lab - 8</b></center>    
<pre>

#### import numpy as np

In [3]:
import numpy as np

#### Create an array of 10 zeros

In [4]:
arr = np.zeros(10)
print(arr)

[0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]


Output should be = array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0.])

#### Create an array of 10 ones

In [5]:
arr = np.ones(10)
print(arr)

[1. 1. 1. 1. 1. 1. 1. 1. 1. 1.]


Output should be = array([1., 1., 1., 1., 1., 1., 1., 1., 1., 1.])

#### Create an array of 10 fives

In [6]:
arr = np.ones(10) * 5

print(arr)

[5. 5. 5. 5. 5. 5. 5. 5. 5. 5.]


Output should be = array([5., 5., 5., 5., 5., 5., 5., 5., 5., 5.])

#### Create an array of the integers from 10 to 50

In [8]:
arr = np.arange(10,51)
print(arr)

[10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33
 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50]


Output should be = array([10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26,
       27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43,
       44, 45, 46, 47, 48, 49, 50])

#### Create an array of all the even integers from 10 to 50

In [9]:
arr = np.arange(10,51,2)
print(arr)

[10 12 14 16 18 20 22 24 26 28 30 32 34 36 38 40 42 44 46 48 50]


Output should be = array([10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40, 42,
       44, 46, 48, 50])

#### Create a 3x3 matrix with values ranging from 0 to 8

In [10]:
arr = np.arange(0,9).reshape(3,3)
print(arr)

[[0 1 2]
 [3 4 5]
 [6 7 8]]


Output should be = <br/>
<pre>array([[0, 1, 2],
       [3, 4, 5],
       [6, 7, 8]])</pre>

#### Create a 3x3 identity matrix

In [12]:
arr = np.eye(3)
print(arr)

[[1. 0. 0.]
 [0. 1. 0.]
 [0. 0. 1.]]


Output should be = <br/>
<pre>array([[1., 0., 0.],
       [0., 1., 0.],
       [0., 0., 1.]])</pre>

#### Use NumPy to generate a random number between 0 and 1

In [31]:
arr = np.random.rand(1)
print(arr)

[0.30293048]


#### Use NumPy to generate an array of 25 random numbers sampled from a standard normal distribution

In [58]:
arr = np.random.rand(25).reshape(5,5)
print(arr)

[[0.02130236 0.76601941 0.92447166 0.4182121  0.79813944]
 [0.75147872 0.7935085  0.94660681 0.74466814 0.67024219]
 [0.325008   0.96627666 0.07856453 0.1508876  0.75050282]
 [0.25415223 0.95208545 0.61405125 0.09044949 0.43004305]
 [0.03039949 0.74113386 0.61651504 0.94342453 0.8060024 ]]


#### Create the following matrix:

In [24]:
arr = np.arange(1,101).reshape(10,10)/100
print(arr)

[[0.01 0.02 0.03 0.04 0.05 0.06 0.07 0.08 0.09 0.1 ]
 [0.11 0.12 0.13 0.14 0.15 0.16 0.17 0.18 0.19 0.2 ]
 [0.21 0.22 0.23 0.24 0.25 0.26 0.27 0.28 0.29 0.3 ]
 [0.31 0.32 0.33 0.34 0.35 0.36 0.37 0.38 0.39 0.4 ]
 [0.41 0.42 0.43 0.44 0.45 0.46 0.47 0.48 0.49 0.5 ]
 [0.51 0.52 0.53 0.54 0.55 0.56 0.57 0.58 0.59 0.6 ]
 [0.61 0.62 0.63 0.64 0.65 0.66 0.67 0.68 0.69 0.7 ]
 [0.71 0.72 0.73 0.74 0.75 0.76 0.77 0.78 0.79 0.8 ]
 [0.81 0.82 0.83 0.84 0.85 0.86 0.87 0.88 0.89 0.9 ]
 [0.91 0.92 0.93 0.94 0.95 0.96 0.97 0.98 0.99 1.  ]]


Output should be = <br/>[[0.01, 0.02, 0.03, 0.04, 0.05, 0.06, 0.07, 0.08, 0.09, 0.1 ],<br/>
       [0.11, 0.12, 0.13, 0.14, 0.15, 0.16, 0.17, 0.18, 0.19, 0.2 ],<br/>
       [0.21, 0.22, 0.23, 0.24, 0.25, 0.26, 0.27, 0.28, 0.29, 0.3 ],<br/>
       [0.31, 0.32, 0.33, 0.34, 0.35, 0.36, 0.37, 0.38, 0.39, 0.4 ],<br/>
       [0.41, 0.42, 0.43, 0.44, 0.45, 0.46, 0.47, 0.48, 0.49, 0.5 ],<br/>
       [0.51, 0.52, 0.53, 0.54, 0.55, 0.56, 0.57, 0.58, 0.59, 0.6 ],<br/>
       [0.61, 0.62, 0.63, 0.64, 0.65, 0.66, 0.67, 0.68, 0.69, 0.7 ],<br/>
       [0.71, 0.72, 0.73, 0.74, 0.75, 0.76, 0.77, 0.78, 0.79, 0.8 ],<br/>
       [0.81, 0.82, 0.83, 0.84, 0.85, 0.86, 0.87, 0.88, 0.89, 0.9 ],<br/>
       [0.91, 0.92, 0.93, 0.94, 0.95, 0.96, 0.97, 0.98, 0.99, 1.  ]]

#### Create an array of 20 linearly spaced points between 0 and 1:

In [57]:
arr = np.linspace(0,1,20).reshape(4,5)
print(arr)

[[0.         0.05263158 0.10526316 0.15789474 0.21052632]
 [0.26315789 0.31578947 0.36842105 0.42105263 0.47368421]
 [0.52631579 0.57894737 0.63157895 0.68421053 0.73684211]
 [0.78947368 0.84210526 0.89473684 0.94736842 1.        ]]


array([0.        , 0.05263158, 0.10526316, 0.15789474, 0.21052632,
       0.26315789, 0.31578947, 0.36842105, 0.42105263, 0.47368421,
       0.52631579, 0.57894737, 0.63157895, 0.68421053, 0.73684211,
       0.78947368, 0.84210526, 0.89473684, 0.94736842, 1.        ])

## Numpy Indexing and Slicing


In [64]:
arr = np.arange(1,26).reshape(5,5)
print(arr)

[[ 1  2  3  4  5]
 [ 6  7  8  9 10]
 [11 12 13 14 15]
 [16 17 18 19 20]
 [21 22 23 24 25]]


Output should be = <br/>
<pre>array([[ 1,  2,  3,  4,  5],
       [ 6,  7,  8,  9, 10],
       [11, 12, 13, 14, 15],
       [16, 17, 18, 19, 20],
       [21, 22, 23, 24, 25]])</pre>

In [69]:
arr[2:,1:]

array([[12, 13, 14, 15],
       [17, 18, 19, 20],
       [22, 23, 24, 25]])

Output should be =
<pre>array([[12, 13, 14, 15],
       [17, 18, 19, 20],
       [22, 23, 24, 25]])</pre>

In [71]:
arr[3,4]

20

Output should be = 20 (element at specific index)

In [83]:
arr[:3,1:2]

array([[ 2],
       [ 7],
       [12]])

Output should be = <br/>
<pre>array([[ 2],
       [ 7],
       [12]])</pre>

In [84]:
arr[4:,0:]

array([[21, 22, 23, 24, 25]])

Output should be = array([21, 22, 23, 24, 25])

In [85]:
arr[3:,0:]

array([[16, 17, 18, 19, 20],
       [21, 22, 23, 24, 25]])

Output should be =  <br/>
<pre>array([[16, 17, 18, 19, 20],
       [21, 22, 23, 24, 25]])</pre>

In [87]:
arr1 = arr[(arr % 3 == 0)&(arr % 5 != 0)]
print(arr1)

[ 3  6  9 12 18 21 24]


Print all the number which are divisible by 3 but not by 5<br/>
Output should be = [ 3  6  9 12 18 21 24]

### Now do the following

#### Get the sum of all the values in mat

In [94]:
arr1 = np.sum(arr)
print(arr1)

325


Output should be = 325

#### Get the standard deviation of the values in mat

In [95]:
arr1 = np.std(arr)
print(arr1)

7.211102550927978


Output should be = 7.211102550927978

#### Get the sum of all the columns in mat

In [93]:
arr1 = sum(arr)
print(arr1)

[55 60 65 70 75]


Output should be = array([55, 60, 65, 70, 75])

# Nice work, Keep the spark alive

================= End of Assignemnt ==================